src/Field.lua

```diff
@@ -0,0 +1,32 @@
+local Runtime = require(script.Parent.Runtime)
+
+local ContextKey = Runtime.createContext("Field")
+
+local Field = {
+	FieldValueType = {
+		TEXTBOX = 1,
+		CHECKBOX = 2,
+		TEXTLABEL = 3,
+	},
+}
+
+local defaultFieldOptions = {
+	depth = 0,
+}
+
+function Field.getOptions()
+	return Runtime.useContext(ContextKey) or defaultFieldOptions
+end
+
+function Field.setOptions(optionsFragment)
+	local existing = Runtime.useContext(ContextKey) or defaultFieldOptions
+	local new = table.clone(existing)
+
+	for key, value in pairs(optionsFragment) do
+		new[key] = value
+	end
+
+	Runtime.provideContext(ContextKey, new)
+end
+
+return Field
```

Comment:
This file doesn't feel like the right solution to me. Can we instead just use Context directly inside of the widgets, or pass options to each one?

Calling `Field.setOptions` just doesn't feel like the right kind of API

Reply:
Initially I wanted to use Context directly inside of the widgets, but I decided to move it out to avoid props drilling especially for the `minCellSize` option. `minCellSize` is updated by the dictionary container and is based on the [absolute content size of the container](https://github.com/metrowaii/plasma/blob/implement-dictionary-widget/src/widgets/dictionary.lua#L65-L69). However, I could invert the widgets to solve this problem. 

Instead of "vector3 is a field" what if I did "field can be a vector3 or a cframe or a string, or a boolean, etc". The dictionary container would only render a field widget, and then the field widget would render another widget based on the type of the value. Something similar to [this](https://github.com/metrowaii/plasma/blob/implement-dictionary-widget/src/widgets/dictionary.lua#L14-L15), but instead the logic would be moved to the field. What do you think?

Reply:
Context is what should prevent prop drilling though. You only need to inject props via context once, and you can get them later in the tree when needed

Reply:
I should have said "I decided to move Context out to **help** avoid props drilling". Without Context, I was passing in the `minCellSize` option from the parent container all the way down to the underlying field widget. To fix this, I moved the Context outside of baseField, and into the Field module that you see right now.

Reply:
On second thought, what do you think of having a config file for each data type, and then having the field widget consume the config and build the view around it?

Reply:
I think what makes the most sense is for the field widget to just accept children which can then be whatever is needed (checkbox, text input, etc)